convert2rhel/checks.py

```diff
@@ -64,6 +65,7 @@ def check_uefi():
     logger.debug("Converting BIOS system")
 
 
+@available_for_skipping(aliases=("tainted", "tainted_kernel", "tainted_kmods"))
```

Comment:
This approach wouldn't solve OAMG-4533 in a way that still warns the user. We'd be completely blind to any kernel modules that is unsupported or tainted. The only warning we get is that the method is skipped. It would be to our advantage to still print kmods but just changing it to a warning as opposed to critical

Would it be better if we have a decorator that instead just catches the SysExit? Or get another variable like OAMG-4533 discusses with `CONVERT2RHEL_DEVEL_KMOD_CHECK_WARN_ONLY`

Reply:
In terms of `Allow overriding the kmod inhibitor` it is solved, however it is not verobose, you're right.
I suggest introducing some special option into `available_for_skipping` decorator as a separate MR. The current concept is very flexible for extending.

Reply:
You've a got a good point. I'd also prefer the switch to not block the conversion but still run the check. Even for the purpose of integration testing it's better to run all the code and not skip the checks all together.

But mere catching SystemExit would not solve it since there's a couple of logger.critical() calls in checks.py which are supposed to stop the conversion when something unexpected happens.

Reply:
In other words, it seems to me that the "skip" feature is just an alternative solution to "warn only", both having the same goal.
And the only disadvantage of the "warn only" is that the conversion would take a bit longer. Otherwise it has more advantages - the user gets more information and more of the c2r code is being tested.

My suggestion would be:

In the check functions, instead of calling logger.critical(msg) for inhibiting the conversion, call a new function `inhibit_conversion(msg, warn_only_keyword)`, that would either:

- stop the conversion (the default), or
- print the message just as a warning when `CONVERT2RHEL_DEVEL_WARN_ONLY=override_keyword`